### DISCUSSION TITLE
Keep context values when hot reloading

### DISCUSSION THREAD

**Discussion by PhanDungTri** [remix-run/remix#1543] [Q&A] [answered]
My Remix app is using context to share values between components, but there is a disadvantage that when I change code then Remix does a hot reloading, the current values of the context is lost (reset to initial values). It makes me keep reproducing many things just to test a new small piece of code, it's a waste of time. How can I keep the state of the context when my app is hot reloaded?

**Comment by sairajchouhan** [comment, marked as answer]
I too had this problem and I often change the initial state in my `useState`  (context in your case) to what I prefer to start from, I know it's not ideal but Remix does not support Hot Module Replacement (yet).

**Reply by PhanDungTri** [reply]
Yeah, I thought of that workaround, I just want to know if there is any solution for this problem.

**Reply by sairajchouhan** [reply]
I guess no for now

**Reply by sergiodxa** [reply]
And they have stated they don't plan to add HMR or Fast Refresh to Remix soon because it will cause inconsistency issues when updating your server-side code that you will not see until in production, by reloading the whole page loaders run again and you can see latest code with the latest data.

One way to solve the issue with having to do many actions to reach the state you want could be to write tests in Cypress, that way when you change the code the test run again and it will do everything for you.

**Reply by XYCaptain** [reply]
> 

Could suppoert Hot Module Replacement for ClientOnly code?
